src/OpenTelemetry.Api/DistributedContext/DistributedContextEntry.cs

```diff
@@ -41,8 +41,26 @@ public DistributedContextEntry(string key, string value)
         /// <param name="metadata">Entry metadata.</param>
         public DistributedContextEntry(string key, string value, in EntryMetadata metadata)
         {
-            this.Key = key ?? throw new ArgumentNullException(nameof(key));
-            this.Value = value ?? throw new ArgumentNullException(nameof(value));
+            if (key == null)
+            {
+                this.Key = string.Empty;
```

Comment:
is dictionary fine with empty string keys? Maybe we can use `"(null)"` instead?

Reply:
DistributedcontextEntries are stored in the list
empty string is typical representation of null in c#, also we use empty string as fallbacks everywhere else. why here it should be different?